tests/store/artifact/test_ftp_artifact_repo.py

```diff
@@ -211,32 +211,99 @@ def test_log_artifact_multiple_calls(ftp_mock, tmpdir):
     assert ftp_mock.storbinary.call_args_list[0][0][0] == "STOR test2.txt"
 
 
+def __posixpath_parents(pathname, root):
+    parents = []
+    parent, _ = posixpath.split(pathname)
+    root = posixpath.normpath(root)
+    parents.append(parent)
+    while parent != "/" and parent != root:
+        parent, _ = posixpath.split(parent)
+        parents.append(parent)
+    return parents
+
+
 @pytest.mark.parametrize("artifact_path", [None, "dir", "dir1/dir2"])
 def test_log_artifacts(artifact_path, ftp_mock, tmpdir):
-    repo = FTPArtifactRepository("ftp://test_ftp/some/path")
+    # Setup FTP mock.
+    dest_path_root = "/some/path"
+    repo = FTPArtifactRepository("ftp://test_ftp" + dest_path_root)
 
     repo.get_ftp_client = MagicMock()
     call_mock = MagicMock(return_value=ftp_mock)
     repo.get_ftp_client.return_value = MagicMock(__enter__=call_mock)
 
+    dir_created = set([dest_path_root])
+    file_created = set()
+    cwd_history = []
+
+    def mkd_mock(pathname):
+        assert pathname == posixpath.abspath(pathname), "mock not implemented"
```

Comment:
`"mock not implemented"` doesn't seem to be an appropriate message for the assert condition `pathname == posixpath.abspath(pathname)`.